examples/integrations/hf_trainer_api/deepspeed.yaml

```diff
@@ -0,0 +1,36 @@
+name: vision transformer deepspeed stage 1
+debug: false
+environment:
+    environment_variables:
+        - NCCL_DEBUG=INFO
+        # You may need to modify this to match your network configuration.
+        - NCCL_SOCKET_IFNAME=ens,eth,ib
+    image:
+      gpu: determinedai/environments:cuda-11.3-pytorch-1.10-tf-2.8-deepspeed-0.7.0-gpu-9119094
+resources:
+  slots_per_trial: 4
```

Comment:
did you test with nodes with 2 gpus each?  otherwise not sure if setting would test model shards split across multiple nodes

Reply:
Short answer: I run the deepspeed config for stage 3 on 2 nodes x 4 GPUs, and I see all files being stored in the checkpoint, and deepspeed resumes without any warnings.

Long answer: before saving the data from all nodes, deepspeed also "worked" but complained in the logs that some states (e.g., model/optimizer) could not be loaded, and then it just moved forward as if nothing happened. After saving sharded data, the warning disappeared, and when running with 4 nodes x 1 GPU, all 4 nodes said they loaded sharded data. 
Now, with 2 nodes x 4 GPUs setting, I see that local_rank=0/4 loads itself and happily announces loading 8 zero partition checkpoints (4 for model and 4 for optimizer). Also, deepspeed does not show any warnings, and logs look fine. Imo works.